src/backend/joanie/core/api/client.py

```diff
@@ -951,4 +952,11 @@ def get_queryset(self):
             if self.request.auth
             else self.request.user.username
         )
-        return models.Contract.objects.filter(order__owner__username=username)
+        qs = models.Contract.objects.filter(
+            Q(
+                order__organization__accesses__user__username=username,
+                order__organization__accesses__role__in=[enums.OWNER, enums.ADMIN],
+            )
+            | Q(order__owner__username=username)
+        ).prefetch_related("order", "order__organization", "order__course")
```

Comment:
the order may be related to an enrollment instead of a course, should we also take care of this?